src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -52,6 +53,15 @@ util::geo::I32Box SpatialJoinAlgorithms::libspatialjoinParse(
       const auto& p = id.getGeoPoint();
       parser.parsePoint(util::geo::DPoint(p.getLng(), p.getLat()), row,
                         leftOrRightSide);
+    } else if (id.getDatatype() == Datatype::LocalVocabIndex) {
+      const auto& literalOrIri = result->localVocab()
+                                     .getWord(id.getLocalVocabIndex())
+                                     .asLiteralOrIri();
```

Comment:
```suggestion
      const auto& literalOrIri = *id.getLocalVocabIndex();
```
(We just encode raw pointers into the id, the local vocab is just for keeping the IDs alive).